File: packages/shared/lib/src/models/vehicle.dart
class Vehicle {
  final String id;
  final String plateNumber;
  final String model;
  final String type;
  final double? maxWeight;
  final double? maxVolume;
  final String? branchId;
  final bool isAvailable;
  final DateTime createdAt;

  Vehicle({
    required this.id,
    required this.plateNumber,
    required this.model,
    required this.type,
    this.maxWeight,
    this.maxVolume,
    this.branchId,
    required this.isAvailable,
    required this.createdAt,
  });

  factory Vehicle.fromJson(Map<String, dynamic> json) {
    return Vehicle(
      id: json['id'] as String,
      plateNumber: json['plate_number'] as String,
      model: json['model'] as String,
      type: json['type'] as String,
      maxWeight: (json['max_weight'] as num?)?.toDouble(),
      maxVolume: (json['max_volume'] as num?)?.toDouble(),
      branchId: json['branch_id'] as String?,
      isAvailable: json['is_available'] as bool? ?? true,
      createdAt: DateTime.parse(json['created_at'] as String),
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'plate_number': plateNumber,
      'model': model,
      'type': type,
      'max_weight': maxWeight,
      'max_volume': maxVolume,
      'branch_id': branchId,
      'is_available': isAvailable,
      'created_at': createdAt.toIso8601String(),
    };
  }
}

File: backend/api/app/models/vehicle.py
import uuid
from datetime import datetime

from sqlalchemy import Boolean, DateTime, Float, ForeignKey, String, func
from app.database import GUID
from sqlalchemy.orm import Mapped, mapped_column

from app.database import Base


class Vehicle(Base):
    __tablename__ = "vehicles"

    id: Mapped[uuid.UUID] = mapped_column(GUID(), primary_key=True, default=uuid.uuid4)
    plate_number: Mapped[str] = mapped_column(String(50), nullable=False, unique=True)
    model: Mapped[str] = mapped_column(String(255), nullable=False)
    type: Mapped[str] = mapped_column(String(100), nullable=False)
    max_weight: Mapped[float | None] = mapped_column(Float, nullable=True)
    max_volume: Mapped[float | None] = mapped_column(Float, nullable=True)
    branch_id: Mapped[uuid.UUID | None] = mapped_column(GUID(), ForeignKey("branches.id"), nullable=True)
    is_available: Mapped[bool] = mapped_column(Boolean, default=True)
    created_at: Mapped[datetime] = mapped_column(DateTime(timezone=True), server_default=func.now())
